libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/internal/route/line/MapboxRouteLineUtils.kt

```diff
@@ -616,29 +614,34 @@ internal object MapboxRouteLineUtils {
         return ranges
     }
 
-    private fun getIntersectionsWithGeometryIndex(steps: List<LegStep>?): List<StepIntersection> {
-        return steps
+    private fun getRoadClassArray(
+        steps: List<LegStep>?
+    ): Array<String?> {
+        val intersectionsWithGeometryIndex = steps
             ?.mapNotNull { it.intersections() }
             ?.flatten()
             ?.filter {
                 it.geometryIndex() != null
             }?.toList() ?: listOf()
-    }
 
-    private fun getRoadClassArray(
-        intersectionsWithGeometryIndex: List<StepIntersection>
-    ): Array<String?> {
-        return if (intersectionsWithGeometryIndex.isNotEmpty()) {
-            arrayOfNulls<String>(
-                intersectionsWithGeometryIndex.last().geometryIndex()!! + 1
-            ).apply {
-                intersectionsWithGeometryIndex.forEach {
-                    this[it.geometryIndex()!!] = it.mapboxStreetsV8()?.roadClass()
-                        ?: "intersection_without_class_fallback"
+        return when (intersectionsWithGeometryIndex.isNotEmpty()) {
+            true -> {
+                arrayOfNulls<String>(
+                    intersectionsWithGeometryIndex.last().geometryIndex()!! + 1
+                ).apply {
+                    intersectionsWithGeometryIndex.forEach {
+                        val roadClass: String =
+                            it.mapboxStreetsV8()?.roadClass()
+                                ?: "intersection_without_class_fallback"
+                        if (it.geometryIndex()!! < this.size) {
+                            this[it.geometryIndex()!!] = roadClass
+                        }
```

Comment:
It's good that we can fail gracefully but I'm worried that because of this developers will have a very little chance to catch the data problem and fix it. Is there a log or some other mechanism that we can add to surface the problem?

Also, if the API contract is that the arrays have to have equal sizes, then I would even be fine saying that we won't fix this issue.

Reply:
I'm not aware of any `DirectionsRoute` JSON contract anywhere.  This function is only used for road classes. AFIK we only have one customer that's using road classes.  I don't like the `!!` in the code so the originally reported issue aside I'd prefer to have a more defensive function.

Reply:
I have in mind a higher level Directions API contract - https://docs.mapbox.com/api/navigation/directions/. Without the arrays of geometry index and annotations being the same, we would not be able to connect the annotation's data to the segment of a route that it's on. If that contract is broken, the data is somewhat useless for us so it makes sense for me to throw or at least aggressively log in these situations.

Reply:
I can add a logging statement.